src/components/helpers/filterHelper.js

```diff
@@ -3,32 +3,28 @@ import { stringify as queryStringStringify } from "query-string";
 const filterHelper = (() => {
   return {
     isObject(object) {
-      return object !== null && typeof object === "object";
+      return !object && typeof object === "object";
     },
+    
     compareObjects(objFirst, objSecond) {
       const keysObjectFirst = Object.keys(objFirst);
       const keysObjectSecond = Object.keys(objSecond);
 
-      if (keysObjectFirst.length !== keysObjectSecond.length) {
-        return false;
-      }
-
-      for (const key of keysObjectFirst) {
+      keysObjectFirst.map(key => {
         const valObjectFirst = objFirst[key];
         const valObjectSecond = objSecond[key];
         const areObjects =
           this.isObject(valObjectFirst) && this.isObject(valObjectSecond);
-        if (
-          (areObjects &&
-            !this.compareObjects(valObjectFirst, valObjectSecond)) ||
+        const compareObjectsKeys =
+        keysObjectFirst.length !== keysObjectSecond.length;
+         
+          return (areObjects && !compareObjectsKeys) ||
           (!areObjects && valObjectFirst !== valObjectSecond)
-        ) {
-          return false;
-        }
-      }
+      });
 
```

Comment:
are you sure about it?

Reply:
I understood my not correct decision to solve the issue. Thanks.